stanza/text/vocab.py

```diff
@@ -7,65 +7,73 @@
 
 
 class Vocab(object):
-    """
-    An abstraction for a vocabulary object that maps between words and integer indices.
-    """
+    """Defines a bijection between N words and the integers 0 through N-1."""
 
-    def __init__(self, unk=''):
-        self.word2index = {}
-        self.index2word = []
-        self.counts = Counter()
-        self.unk = unk
+    def __init__(self, unk):
+        """Construct a Vocab object.
 
-        if self.unk:
-            self.add(self.unk, 0)
-
-    def clear_counts(self):
+        Args:
+            unk: a string to represent the unknown word (UNK).
         """
-        removes counts for all tokens
-        """
-        self.counts.clear()
+        self.__word2index = OrderedDict()
+        self.__counts = Counter()
+        self.__unk = unk
+
+        # assign an index for UNK
+        self.add(self.__unk, count=0)
+
+    def __iter__(self):
+        return iter(self.__word2index)
 
     def __repr__(self):
-        return str(self.word2index)
+        """Represent Vocab as a dictionary from words to indices."""
+        return str(self.__word2index)
+
+    def __str__(self):
+        return 'Vocab(%d words)' % len(self.__word2index)
 
     def __len__(self):
-        return len(self.index2word)
+        """Get total number of entries in vocab."""
+        return len(self.__word2index)
 
     def __getitem__(self, word):
-        if self.unk:
-            return self.word2index.get(word, self.word2index[self.unk])
-        else:
-            return self.word2index[word]
+        """Get the index for a word.
+
+        If the word is unknown, the index for UNK is returned.
+        """
+        try:
```

Comment:
Maybe we can do without exception handling which is very slow?
eg. `return self.__word2index.get(word, self.__unk)`